salt/tls/init.sls

```diff
@@ -25,3 +44,91 @@ ssl-cert:
     - require:
       - pkg: ssl-cert
 {% endfor %}
+
+{% if salt["match.compound"](pillar["roles"]["salt-master"]["pattern"]) %}
+# HTTP-validated domains
+{% for domain in [
+    'pypa.io',
```

Comment:
Lists like these are always a good sign that something should be in pillar rather than a state.

I'd suggest moving this list into a subkey of https://github.com/python/psf-salt/blob/7fcd4a36b615cb629f3afa60912f0b128b99e4f9/pillar/base/tls.sls#L4

Maybe `acme_certs`.

Something like:

```
tls:
  acme_certs:
    example.com:
      validation: http
      roles:
      - loadbalancer
      - example
      additional_sans:
      - www.example.com
```

This would also simplify the logic in your pillar extension to allow us to determine where certs are validated (once dns is supported) and distributed.